default.nix

```diff
@@ -1,9 +1,9 @@
-{ rev ? "541a3ca27c9a8220b46f4feb7dd8e94336a77f42", # nixpkgs master
+{ rev ? "a7855f2235a1876f97473a76151fec2afa02b287", # nixpkgs master. Keep up to date with flake.lock
```

Comment:
I'd like to be more specific here and change this to:
```
# nixpkgs master. Keep up to date with "nixpkgs">"locked">"rev" in flake.lock
```

Reply:
Do you think it'd be fine to actually read this property from the lockfile? We can parse it as JSON and extract the rev so that it's always up to date.

Also, did you consider removing these defaults? (with flakes as the only entrypoint to the build)

Reply:
> Do you think it'd be fine to actually read this property from the lockfile?

I like to keep the `default.nix` file self-contained in preparation for a release on nixpkgs.

> Also, did you consider removing these defaults?

With defaults I believe you mean `default.nix`. Yes, we only keep this around because I believe they are mandatory to release a roc package on nixpkgs. As soon as we can use flakes there we can delete default.nix.

Reply:
>  in preparation for a release on nixpkgs

Thought about this too :) makes sense!

> With defaults I believe you mean default.nix

I meant the `? "..."` syntax - not sure what it's called in nix exactly, but my point was "default values of the function arguments". But my question is answered by the above too.